ishop/views.py

```diff
@@ -1,3 +1,179 @@
-from django.shortcuts import render
+from django.contrib.auth import get_user_model, login
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.contrib.auth.views import LoginView, LogoutView
+from django.shortcuts import redirect
+from django.utils import timezone
+from django.views.generic import ListView, CreateView, UpdateView
+from django.contrib import messages
 
-# Create your views here.
+from ishop.forms import CustomUserCreationForm
+from ishop.models import Good, ShopUser, Purchase, Refund
+
+
+class SuperUserRequiredMixin(LoginRequiredMixin, UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_superuser
+
+
+class GoodsListView(ListView):
+    paginate_by = 10
+    http_method_names = ['post', 'get']
+    template_name = 'goods_list.html'
+    queryset = Good.objects.filter(in_stock__gt=0)
+    extra_context = {'title': 'Online shop'}
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Add post to the List View is not a good practice, possible but it looks like a broking of SOLID. The best way 2 different views for 2 different actions

Reply:
Before creating a pull request I had 2 more View: 
 1 - for User after buying (this post-function was there)
 2 - for Admin after approving refund (again, with post logic)
but the pages which they returned don't have any useful sense - only to show the results 
I went a bit forward and added messages on the project - after that these 2 additional pages became even more unuseful...

If I don't want to have many almost blank pages - maybe I should create RedirectView, put there all logic (solid!), and return to the same page?
Would it be better?

Reply:
upd:
something like CreateView + success url could help, yes?

Reply:
done on base View

Reply:
In the main Idea it's a create of purchase, why not create view?